### PR TITLE
Add support for optional footer in Reveal

### PR DESCRIPTION
**Pull request by mthmulders**
Following the discussion in #3, here's a PR that adds support for passing in a path to a custom footer.

### CODE REVIEW

**Comment by riezebosch** [comment]
```
Good job! At least you didn't create new surviving mutants.
----------------|---------|----------|-----------|------------|----------|---------|
File            | % score | # killed | # timeout | # survived | # no cov | # error |
----------------|---------|----------|-----------|------------|----------|---------|
All files       |  100.00 |      102 |         0 |          0 |        0 |      42 |
 plugins        |  100.00 |       67 |         0 |          0 |        0 |      33 |
  CustomCss.ts  |  100.00 |        5 |         0 |          0 |        0 |       2 |
  Footer.ts     |  100.00 |        6 |         0 |          0 |        0 |      10 |
  Highlight.ts  |  100.00 |       21 |         0 |          0 |        0 |       4 |
  Img.ts        |  100.00 |        2 |         0 |          0 |        0 |       1 |
  Reveal.ts     |  100.00 |        4 |         0 |          0 |        0 |       1 |
  Slides.ts     |  100.00 |       13 |         0 |          0 |        0 |       9 |
  StaticBase.ts |  100.00 |        0 |         0 |          0 |        0 |       1 |
  Template.ts   |  100.00 |        2 |         0 |          0 |        0 |       1 |
  Theme.ts      |  100.00 |       14 |         0 |          0 |        0 |       4 |
 Index.ts       |  100.00 |        2 |         0 |          0 |        0 |       1 |
 Server.ts      |  100.00 |       15 |         0 |          0 |        0 |       0 |
 SlideObject.ts |  100.00 |       18 |         0 |          0 |        0 |       8 |
----------------|---------|----------|-----------|------------|----------|---------|
```

**Comment by mthmulders** [comment]
Did you know you can get a [badge](https://dashboard.stryker-mutator.io/) for that?

**Comment by riezebosch** [comment]
Cool but I'm no longer the owner of this account and only have access to this specific repo...

File: test/test_data/footer.html [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<img src="acme.png" alt="ACME Logo">
```
Comment (riezebosch) [inline test/test_data/footer.html:1]: This file is not present and the url should be "/img/acme.png". Probably we should add a integration test that checks for `400`'s.

**Review by riezebosch — COMMENTED** [review]
almost done

File: test/test_data/footer.html (hunk 1 above)
Comment (mthmulders) [reply test/test_data/footer.html:1]: Well, it doesn't have to exist - I'm just using it as an example that you could add more than one DOM element. Could be a `<b>whatever</b>`, too; didn't want to pollute the repo with stuff that doesn't belong to kc-serve.

Comment (riezebosch) [reply test/test_data/footer.html:1]: Yeah, but you included a picture.jpg and not an acme.png

**Comment by mthmulders** [comment]
> Cool but I'm no longer the owner of this account and only have access to this specific repo...

Then who is holding the account? Otherwise, this repo (or even the account) would better off being closed.

By the way, does that imply you are also not able to create a new release? Again, who would be?

**Comment by riezebosch** [comment]
Yes, I can. The account is maintained by members of KC and I have access to this repo and the npm package.

**Comment by riezebosch** [comment]
On second thought I think your contributions will even be better if you don't specify it on the command line but scan the magic folder `footer` like the other plugins do. That way it even integrates more easy with the `@infosupport/kc-cli` that is used by you and your colleagues.

**Comment by mthmulders** [comment]
> On second thought I think your contributions will even be better if you don't specify it on the command line but scan the magic folder `footer` like the other plugins do.

It could work, too, although I would consider it less elegant. For example, what to do if there are more than one files in that folder? I think I'd prefer a fixed `footer.html` in that case - so that you can be sure it'll always be exactly one file.

How about that?

**Comment by riezebosch** [comment]
Just like the markdown files: always create footer section and add the content of the files in (natural sorted) order. That would be more in line with the `css`, `script` and `slides` folders.

**Comment by mthmulders** [comment]
I understand the idea, but I would consider it a bit over-complicated. Usually, a footer contains just an image and a few words (company name, hashtag, that kind of stuff). For my goals, one file (either static or parameterised) would be more than enough.

**Comment by mthmulders** [comment]
I noticed I made a mistake in the implementation. What I wanted to achieve is this:

    <div class="reveal">
      <div class="slides">
        
      </div>
      <div class="footer">
        <p>Copyright ©2018 MyCompany, Inc.</p>
      </div>
    </div>

But the current implementation does this:

    <div class="reveal">
      <div class="slides">
        
      </div>
    </div>
    <div class="footer">
      <p>Copyright ©2018 MyCompany, Inc.</p>
    </div>

Note how the `div.footer` lives at the same level at `div.reveal`, instead of being a child element of it.

**Comment by mthmulders** [comment]
> On second thought I think your contributions will even be better if you don't specify it on the command line but scan the magic folder footer like the other plugins do. 

Did you mean to give a suggestion for further development? Or should I read: "I'm not merging this until..."?

**Comment by mthmulders** [comment]
In hindsight, passing a template to a template doesn't feel right. I'll rework that later today.

**Comment by riezebosch** [comment]
I was trying to find the time to test out my idea on top of your current contribution. Had some trouble pulling your changes because it doesn't seem to be up-to-date with my master branch.

**Review by riezebosch — COMMENTED** [review]
plugin = for Server, to host files
template = output, manipulate html

what you want is extend the Slides template

**Comment by mthmulders** [comment]
I think I have done just that? I've extended the slides template with an additional DOM element (`<div class="footer">`), while the plugin determines _what_ file should be used/served to fill that element.

**Comment by riezebosch** [comment]
All plugins derive from `StaticBase` because they also serve the purpose of resolving files because they are aware of the location. Your footer is not, because these aren't to be served but rather embedded. I guess the plugins are currently more of a two-trick-pony and resolving and serving could be split.

**Comment by mthmulders** [comment]
I hadn't noticed yet, but digging a little further through the source code reveals a pony that does two tricks. The `CustomCss` plugin seems a nice example. I'd feel a little uncomfortable refactoring that right now, though....

**Comment by riezebosch** [comment]
I've included your changes but did a rebase on master first. See [your version](https://www.npmjs.com/package/@infosupport/kc-serve) being published.

Expect the command-line option to be dropped in the feature though, since I aim for a zero-conf setup. Will refactor the resolving part out and scan the `/footer` folder from the slides template.

Sorry it took so long.

**Comment by mthmulders** [comment]
Thanks for merging this in!

**Pull request closed without merging**